packaging/technical-addon/Splunk_TA_otel/windows_x86_64/bin/Splunk_TA_otel_utils.ps1

```diff
@@ -1,63 +1,85 @@
 $otelProcessName=$args[0]
-$splunkTAOtelLogFileName=$args[1]
-$splunkTAOtelLogDir=Split-Path -Path $splunkTAOtelLogFileName
-$splunkTAPSLogFile=Join-Path -Path $splunkTAOtelLogDir -ChildPath Splunk_TA_otelutils.log
 
 function otelLogWrite
 {
    Param ([string]$logstring)
    $timestamp = Get-Date -Format "yyyy-MM-dd HH:mm:ss.fff K"
-   Add-content $splunkTAPSLogFile -value "$timestamp $logstring"
+
+   # Write-Warning is much less verbose than Write-Error and the other Write-* may end up
+   # not being captured by the parent process (expected to be the cmd that launched the script)
+   Write-Warning "$timestamp Splunk_TA_otel_utils.ps1: $logstring"
+}
+
+function getParentProcessId($id)
+{
+	return (Get-WmiObject Win32_Process | Where-Object ProcessId -eq $id).ParentProcessId
+}
+
+function getParentAndGrandParent($id)
+{
+	try {
+		$parentId = getParentProcessId($id)
+		otelLogWrite "INFO Parent process id: $parentId"
+		$grandParentId = getParentProcessId($parentId)
+		otelLogWrite "INFO GrandParent process id: $grandParentId"		
```

Comment:
I see a comment below stating the value of directly using `Write-Error`, but what's the purpose of `otelLogWrite`?

Reply:
Right now, just adding the timestamp.